include/PolygonSimplification.hpp

```diff
@@ -33,62 +33,107 @@
 
 namespace com {
 	namespace geopipe {
-		namespace PolySimp {
-			using CGALKernel = CGAL::Exact_predicates_exact_constructions_kernel;
-			using CGALPoint = CGALKernel::Point_2;
-			using CGALPolygon = CGAL::Polygon_2<CGALKernel>;
+		template<typename Kernel = CGAL::Exact_predicates_exact_constructions_kernel, typename Point = typename Kernel::Point_2>
+		class PolySimp {
+			using CGALKernel = Kernel;
```

Comment:
```suggestion
		namespace PolySimpCustomization { 
			template<class Point>
			struct cgal_infer_point_kernel {
				using type = typename Point::Kernel;
			};
			
			template<class Kernel>
			struct cgal_infer_point_kernel< Point_2<Kernel> > {
				using type = Kernel;
			};
			
			template<class Point>
			using cgal_infer_point_kernel_t = typename cgal_infer_point_kernel<Point>::type;
		}
		template<typename Point = typename CGAL::Exact_predicates_exact_constructions_kernel::Point_2>
		class PolySimp {
			using CGALKernel = PolySimpCustomization::cgal_infer_point_kernel_t<Point>
```

Reply:
Since `Kernel` should be linked to `Point`, we will not allow them to be specified independently. Client code should either implement the given customization API or provide a specialization.